packages/compass/src/main/application.ts

```diff
@@ -230,6 +233,12 @@ class CompassApplication {
         debug('Did not agree to license, quitting app.');
         app.quit();
       },
+      'compass:check-secret-storage-is-available': function () {
+        // Accessing isEncryptionAvailable is not allowed when app is not ready on Windows
+        // https://github.com/electron/electron/issues/33640
+        // But here Compass is already rendered and it is safe to omit `await app.whenReady()`
+        return safeStorage.isEncryptionAvailable();
```

Comment:
Not a blocker, does it work if we instead make this method `async` and have the `await app.whenReady();` in here before we return? That way we aren't waiting for the `whenReady` till it's needed.
Looks like the process communication here already has support for async: https://github.com/mongodb-js/compass/blob/3868044de073e6db26b76b4fa4d716f29699606b/packages/hadron-ipc/src/main.ts#L64

Reply:
We could make the callback of `compass:check-secret-storage-is-available` async and put `await app.whenReady()` inside before returning, but why would you want to do that, if we know that app got ready in main before the rendering process?